docs/ru/http-headers.md

```diff
@@ -0,0 +1,193 @@
+
+Базовым классом для коллекции значений заголовков является `\Yiisoft\Http\Header\Header`. Он подходит для заголовков,
+в которых важно соблюдение порядка значений в заданной последовательности. Для сортируемых списков значений выделена
+отдельная коллекция `\Yiisoft\Http\Header\SortableHeader`, а также специальная коллекция
+`\Yiisoft\Http\Header\AcceptHeader` для семейства заголовков `Accept`. В сортируемых коллекциях значения сразу занимают
+места по порядку согласно параметру `q` или иным правилам, описанным в коллекции.
+
+Рассмотрим пример с заголовком `Forwarded` [RFC7239](https://tools.ietf.org/html/rfc7239).\
+Заголовок представляет собой список из пустых значений с возможными параметрами `by`, `for`, `host` и `proto`:
+
+```
+Forwarded: for=192.0.2.43,for="[2001:db8:cafe::17]",for=unknown
+Forwarded: for=192.0.2.60;proto=http;by=203.0.113.43
+```
+
+Следующий код выводит все значения параметров `for`:
+
+```php
+use \Yiisoft\Http\Header\Value\Forwarded;
+
+/** @var \Psr\Http\Message\ServerRequestInterface $request */
+$header = Forwarded::createHeader()->withValues($request->getHeader('Forwarded'));
+
+foreach ($header->getValues() as $headerValue) {
+    $paramFor = $headerValue->getParams()['for'] ?? null;
+    if ($paramFor === null) {
+        continue;
+    }
+    echo "{$paramFor}\n";
+}
+```
+
+Каждый объект коллекции `Header` привязывается к указанному при создании классу заголовка. В коллекцию одного заголовка
+нельзя передать значения другого заголовка, даже если один из них является наследником другого:
+
+```php
+use \Yiisoft\Http\Header\Value\Date;
+use \Yiisoft\Http\Header\Value\Forwarded;
+
+$header = Forwarded::createHeader();
+
+// Будет брошено исключение
+$header = $header->withValue(new Date(new DateTimeImmutable()));
+```
+
+Для взаимодействия с объектами PSR7, реализующими интерфейс `\Psr\Http\Message\MessageInterface`, в классе `Header`
+предусмотрены методы импорта и экспорта заголовков: `exctract()` и `inject()`
+
+```php
+use \Yiisoft\Http\Header\Value\Allow;
+/** @var \Psr\Http\Message\ServerRequestInterface $request */
+/** @var \Psr\Http\Message\ResponseInterface $response */
+
+// Получить коллекцию значений заголовка Allow из объекта запроса
+$header = Allow::createHeader()->extract($request);
+// Или
+$header = Allow::extract($request);
+
+// Записать заголовки в объект ответа
+$response = $header->inject($response, $replace = true);
+```
+
+## Заголовки
+
+### Date
+
+Вы можете использовать класс `Date` для конвертирования объекта `\DateTimeInterface` в строку формата времени HTTP.
+
+```php
+$date = new \Yiisoft\Http\Header\Value\Date(new DateTimeImmutable('2020-01-01 00:00:00 +0000'));
+
+echo $date;
+// Wed, 01 Jan 2020 00:00:00 GMT
+```
+
+### ETag
+
+```php
+use Yiisoft\Http\Header\Value\Condition\ETag;
+/** @var \Psr\Http\Message\ResponseInterface $response */
+
+$etag = (new ETag())->withTag('56d-9989200-1132c580', true)->inject($response);
```

Comment:
```suggestion
$etag = (new ETag('56d-9989200-1132c580'))->addToResponse($response);
```

What is `true` for?

Reply:
> What is true for?

`public function withTag(string $tag, bool $weak = true): self`

Reply:
`(new ETag('56d-9989200-1132c580'))` also possible
`(new ETag('W\\56d-9989200-1132c580'))` too

Reply:
```php
$etag = (new ETag('56d-9989200-1132c580'))->weak()->addToResponse($response);
```

Reply:
> How about this?

it is possible. Right now you can not set the `weak` value saparated from value.
`weak()` or `withWeak()`?
Shoul i remove the `with` prefix in all immutable methods?

Reply:
`weak()` is alright.